src/objects/map/map.vue

```diff
@@ -0,0 +1,442 @@
+          container: mapConfig.containerId,
+          center: mapConfig.center,
+          zoom: mapConfig.zoom,
+          style: mapConfig.style
+        });
+
+        this.map.addControl(new mapboxgl.NavigationControl());
+        this.map.on('load', () => this.mapLoaded = true);
+      },
+      initializeLayer(layer) {
+        if (this.map.getLayer(layer.name) === undefined && Object.entries(layer.data).length !== 0) {
+          const visibility = layer.name === this.activeLayer ? 'visible' : 'none';
+          const filter = layer.filterBy ? ['in', layer.filterBy, ''] : [];
+          const fill = this.generateFillColor();
+
+          this.map.addSource(layer.name, {
+            'type': 'geojson',
+            'data': layer.data
+          });
+
+          this.map.addLayer({
+            'id': layer.name,
+            'type': 'fill',
+            'source': layer.name,
+            'paint': {
+              'fill-outline-color': fill[0],
+              'fill-color': fill[1],
+              'fill-opacity': 0.7
+            },
+            'layout': {
+              'visibility': visibility
+            }
+          });
+
+          this.map.addLayer({
+            'id': `${layer.name}-highlighted`,
+            'type': 'fill',
+            'source': layer.name,
+            'paint': {
+              'fill-outline-color': fill[0],
+              'fill-color': fill[1],
+              'fill-opacity': 1
+            },
+            'filter': filter,
+            'layout': {
+              'visibility': visibility
+            }
+          });
+
+          this.initializeFilter(layer);
+          this.updateCursorOnHover(layer.name);
+        }
+      },
+      initializePopup(event, layer) {
+        this.config.mapType === 'multi' ? this.popupMultiLayer(event, layer) : this.popupSingleLayer(event, layer);
+      },
+      initializeToggle(toggleList) {
+        this.config.mapType === 'multi' ? this.toggleMultiLayer(toggleList) : this.toggleSingleLayer(toggleList);
+      },
+      initializeFilter(layer) {
+        this.config.mapType === 'multi' ? this.filterMultiLayer(layer) : this.filterSingleLayer(layer);
+      },
+      updateCursorOnHover(layerId) {
+        const map = this.map;
+
+        // change the cursor to a pointer when the mouse is over the layer.
+        map.on('mouseenter', layerId, function () {
+          map.getCanvas().style.cursor = 'pointer';
+        });
+
+        // change the cursor back to the default when it leaves the layer.
+        map.on('mouseleave', layerId, function () {
+          map.getCanvas().style.cursor = '';
+        });
+      },
+      generateFillColor() {
+        // NYCO colors, ['fill-outline', 'fill']
+        const colors = [
+          // black, red
+          ['#000000', '#F2695D'],
+          // black, orange
+          ['#000000', '#FFA133'],
+          // dark blue, gray
+          ['#2F334F', '#ACAEB9'],
+          // blue, orange
+          ['#3155A6', '#F9A137']
+        ];
```

Comment:
@KISS I think this is not a blocker but was curious if you thought there was any way we could externalize the color settings... I think it's ok if it's hard-coded but preferably we can manage it from the config/variables.js file. I suppose we could pass it through the properties.

Reply:
@devowhippit yeah, I considered having the colors set on the `layer` object but wanted to make sure there was proper color contrasting and that the colors matched those set for the library. figured hard-coding the values was simplest, since the layer color has to be set as a property when it's added to Mapbox.

would creating a similar attribute to `colorCombinations` in `variables.js` and then importing `variables.js` into `Maps.js`, so it can be used to set a `color` property on the layer, work better?

Reply:
@KISS if that is possible it sounds like the right path to me!

Reply:
@devowhippit I removed the hard coded values. I couldn't import `variables.js` directly because it has a `package.json` dependency so I created a `colors.js` config instead. I moved `colors` and `colorCombinations` into `colors.js` to keep everything in one place and require `colors.js` in `variables.js`. 

I also updated the repo so we can use markdown and updated the docs. To view the docs you'll need to add the `.html` ext: http://localhost:3000/map.html cc/ @kimpenguin